### PR TITLE
Add Helper to verify if URLError is recoverable

### PR DESCRIPTION
**Pull request by addb**
## Description
From the suggestion by @sbenedicadb added a helper which will help in cleaner implementation for extensions.

#### Sample usage
Before:

```
if let urlError = connection.error as? URLError {
    let urlErrorCode = urlError.code

    // retry for recoverable url error codes
    if NetworkServiceConstants.RECOVERABLE_URL_ERROR_CODES.contains(urlErrorCode) {
        ...
    }
}

```

After:
```
 if let urlError = connection.error as? URLError, urlError.isRecoverable {
                      
}
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: AEPServices/Sources/utility/URLError+Helper.swift [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+*/
+
+import Foundation
+
+extension URLError {
```
Comment (sbenedicadb) [inline AEPServices/Sources/utility/URLError+Helper.swift:15]: would prefer the name of this file be more descriptive of what it's actually doing.

maybe something like `URLError+Recoverable.swift`?

**Review by sbenedicadb — APPROVED** [review]
looks good - just one suggestion about the file name for the extension

File: AEPServices/Tests/utility/URLError+RecoverableTests.swift [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        var recoverableErrors = [
+            URLError.timedOut,
+            URLError.cannotConnectToHost,
+            URLError.networkConnectionLost,
+            URLError.notConnectedToInternet,
+            URLError.dataNotAllowed
+        ]
+        
+        recoverableErrors.forEach { errorCode in
+           XCTAssertTrue(URLError(errorCode).isRecoverable)
```
Comment (kevinlind) [inline AEPServices/Tests/utility/URLError+RecoverableTests.swift:29]: nit: Add message so we know on which `errorCode` the test failed.

File: AEPServices/Tests/utility/URLError+RecoverableTests.swift [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        let unrecoverableErrors = [
+            URLError.cannotFindHost,
+            URLError.secureConnectionFailed,
+            URLError.serverCertificateNotYetValid,
+            URLError.unknown,
+            URLError.badURL
+        ]
+        
+        unrecoverableErrors.forEach { errorCode in
+           XCTAssertFalse(URLError(errorCode).isRecoverable)
```
Comment (kevinlind) [inline AEPServices/Tests/utility/URLError+RecoverableTests.swift:43]: nit: Add message so we know on which `errorCode` the test failed.

**Review by kevinlind — APPROVED** [review]

**Pull request merged**